openpathsampling/engines/openmm/features/instantaneous_temperature.py

```diff
@@ -34,16 +34,24 @@ def instantaneous_temperature(snapshot):
     """
     # TODO: this can be generalized as a feature that works with any
     # snapshot that has features for KE (in units of kB) and n_dofs
-    old_snap = snapshot.engine.current_snapshot
-    snapshot.engine.current_snapshot = snapshot
-    state = snapshot.engine.simulation.context.getState(getEnergy=True)
+
+    # if no engine, error here; don't get caught in try/except below
+    engine = snapshot.engine
+    try:
+        old_snap = engine.current_snapshot
+    except Exception:  # openmm doesn't use a custom exception class yet
+        # Exception: Particle positions have not been set
```

Comment:
This is a **very** broad catch (as you mention in the comments), I might have added
```suggestion
    except Exception as e:  # openmm doesn't use a custom exception class yet
        # Exception: Particle positions have not been set
        if "Particle positions" not in str(e):
            raise e
```
feel free to ignore though

Reply:
Thanks. Not taking this because basically, whatever problem there is in loading current snapshot, that should be ignored here (could be missing velocities, etc.) It means the currently internal snapshot represetnation is not valid snapshot, so overriding it with the snapshot for which we want the temperature should not be a problem. The note is mainly for informational purposes (and I do wish that OpenMM returned some sort of `GetStateError`, instead of super-general `Exception`, but that's an issue for them.)